azavea/climate-change-components
# Initial Project Setup

## Overview

Pulls in several models and services from [azavea/climate-change-lab](https://github.com/azavea/climate-change-lab) so they can be re-used in other Angular 4+ applications.

### Notes
 - Supports AoT bundling with angular-library-builder (`nglb`). I originally tried using `ng-packagr` instead, but that doesn't support the new version of typescript used in `climate-change-lab`.
 - Most files were copied from climate-change-lab models/services, modified to inject in `apiHost` constant and the `apiHttp` service.
 - `ng e2e` doesn't work, but I left the stub implementations generated by the Angular CLI in place in case we need it when we bring in components
 - Right now all exports are exposed in one flat grouping in `src/lib/public_api.ts`, with everything importable from `climate-change-components`. It would be nice for there to be some structure, so you could import from `climate-change-components/api` or perhaps instead `climate-change-components/models`

## Testing Instructions

 * Clone this repository and checkout this branch
 * Run `yarn` to install dependencies
 * `yarn run test` should work
 * `yarn run lint` should work
 * `yarn run build:library` should work and produce a `lib-dist/` folder

Closes #2
